src/Core/Email/EmailDataConfigurator.php

```diff
@@ -26,50 +26,41 @@
 
 namespace PrestaShop\PrestaShop\Core\Email;
 
-use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
-use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Configuration\AbstractMultistoreConfiguration;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * Class EmailDataConfigurator is responsible for configuring email data.
  */
-final class EmailDataConfigurator implements DataConfigurationInterface
+final class EmailDataConfigurator extends AbstractMultistoreConfiguration
```

Comment:
You need to add unit tests for this class Especially regarding the sub configurations for `smtp_config` and `dkim_config` (see comments below about sub option resolvers)